packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/feature_counter/blocs/counter_bloc_extensions.dart

```diff
@@ -13,5 +13,16 @@ extension _CounterExtension on CounterBloc {}
 
 extension _ToError on Stream<Exception> {
   /// TODO: Implement error event-to-state logic
-  Stream<String> toMessage() => map((errorState) => errorState.toString());
+  Stream<String> toMessage() =>
+      map((errorState) => errorState.toString().substring(10));
+
+  /// Filter exceptions and present custom message, thrown from the API
+  Stream<Exception> mapFromDio() => where((exception) =>
+  exception is DioError &&
```

Comment:
As I mentioned previously in one of our calls, `where` ( https://api.flutter.dev/flutter/dart-core/Iterable/where.html ) will filter out any other exceptions that do not match the specified criteria. So if this extension method is used, we won't be able to display any exceptions other that the ones matching the criteria.
Since we just want to parse the contents of the error and display it in a different form, we should use the `map` ( https://api.flutter.dev/flutter/dart-core/Iterable/map.html ) function. Inside of it, we can check whether it matches specific criteria and map it to a human-readable error.

Reply:
We use the state, we are using this extension for, for displaying a snack bar. In this case we will show an ugly snack bar when server is not running along to presenting error screen. Because we need to return the same exception if it does not match the if case.

Reply:
Since we are using this extension in `_mapToErrorsState` in the current state it will not show any errors other than the ones with matching criteria. The `errors` state should be used to present the errors to the user (any errors actually). If needed, we can parse those ugly errors inside the extension method and adjust them to they are readable.

Your current code is preventing other (potential) errors for passing through. Without them, if the user does something wrong, he will never know why it is not working. For this case, we are using a snack bar to provide feedback based on his actions.

Reply:
Have a look at this example below:
```
Stream<Exception> someMethodName() => map((e) {
  if (e is DioError){
    // ...
    return Exception(/* here goes the formated part of the dio error */);
  }
  return e;
});
```

Reply:
Edited.